src/physics/base/Secondary.hh

```diff
@@ -0,0 +1,60 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file Secondary.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Types.hh"
+#include "ParticleDef.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * New particle created via an Interaction.
+ *
+ * It will be converted into a track using the original track information.
+ */
+struct Secondary
```

Comment:
The parentage is an important component.  Is it the intent to have that information available from other data pieces (assumingly reachable from the code dealing with the secondary creation.   Since it is not a full fledge secondary (because of 'not directly' available info) may be the name could be tweaked to reflect that.   Maybe "SecondarySeed" (or something better)?

Reply:
I originally had a "parent ID", but removed it because secondaries are reachable from the Interaction (which is tied to a particular track). I envision the "secondary" here as being the "seed" for a track initializer (which can be also created from a primary).

Reply:
So indeed, I would tweak the name.

Reply:
I'd like to avoid "seed" because of the association with PRNG seed. Since there isn't going to be any other sort of "secondary" (just the same "track initializers" that are created from primaries) I don't think leaving it as Secondary will cause any ambiguity, but I'm open to other suggestions if you disagree.

Reply:
Fair enough, seed is for random numbers, still this is not a full fledged secondary, so may be something like `SecondaryRequest`, `SecondaryPrecursor`, `ProtoSecondary`, etc.. ?

Reply:
@pcanal I see where you're coming from, but I'm going to push back a little. I think (at least in this case) it's better to favor a simple, memorable class name than one that fully describes all the subtleties of the class's usage. I think that like the KN interactor we can revisit name choices as more use cases show up.